lib/policy_machine.rb

```diff
@@ -296,40 +316,49 @@ def bulk_persist
 
   private
 
-    # Raise unless the argument is a policy element.
-    def assert_policy_element_in_machine(arg_pe)
-      unless arg_pe.is_a?(PM::PolicyElement)
-        raise(ArgumentError, "arg must each be a kind of PolicyElement; got #{arg_pe.class.name} instead")
-      end
-      unless arg_pe.policy_machine_uuid == self.uuid
-        raise(ArgumentError, "#{arg_pe.unique_identifier} is not in policy machine with uuid #{self.uuid}")
-      end
+  # Raise unless the argument is a policy element.
+  def assert_policy_element_in_machine(arg_pe)
+    unless arg_pe.is_a?(PM::PolicyElement)
+      raise(ArgumentError, "arg must each be a kind of PolicyElement; got #{arg_pe.class.name} instead")
+    end
+    unless arg_pe.policy_machine_uuid == self.uuid
+      raise(ArgumentError, "#{arg_pe.unique_identifier} is not in policy machine with uuid #{self.uuid}")
     end
+  end
 
-    # According to the NIST spec:  "the triple (u, op, o) is a privilege, iff there
-    # exists an association (ua, ops, oa), such that user u→+ua, op ∈ ops, and o→*oa."
-    # Note:  this method assumes that the caller has already checked that the given operation is in the operation_set
-    # for all associations provided.
-    def is_privilege_single_policy_class(user_or_attribute, object_or_attribute, associations)
-      # does there exist an association (ua, ops, oa), such that user u→+ua, op ∈ ops, and o→*oa?
-      associations.any? do |assoc|
-        user_or_attribute.connected?(assoc.user_attribute) && object_or_attribute.connected?(assoc.object_attribute)
-      end
+  # Raise unless the policy elements are policy elements in different machines.
+  def assert_different_machines(pe, another_pe)
+    if !pe.is_a?(PM::PolicyElement) || !another_pe.is_a?(PM::PolicyElement)
```

Comment:
I would agree if it were just one line, but multi-conditional unless blocks always make my head hurt.